askama_derive/src/generator.rs

```diff
@@ -73,9 +93,66 @@ fn build_template(ast: &syn::DeriveInput) -> Result<String, CompileError> {
     if input.print == Print::Code || input.print == Print::All {
         eprintln!("{code}");
     }
+
+    #[cfg(feature = "cache")]
+    cache_template(&digest, &code)?;
+
     Ok(code)
 }
 
+#[cfg(feature = "cache")]
+fn find_cached_template(digest: &str) -> Option<String> {
+    let root = PathBuf::from(option_env!("OUT_DIR")?);
+    let path = root.join(digest);
+
+    path.exists()
+        .then(|| std::fs::read_to_string(path).ok())
+        .flatten()
+}
+
+#[cfg(feature = "cache")]
+fn cache_template(digest: &str, source: &str) -> Result<(), CompileError> {
+    let root =
+        PathBuf::from(option_env!("OUT_DIR").ok_or(CompileError::from("OUT_DIR is not defined"))?);
```

Comment:
Makes sense. There is potential that someone may be using askama without cargo, but that is kind of at their peril anyway, and it's probably better to have the compile error appear there instead of as a compile error.

Reply:
Wow, very niche case but perhaps worth a callout in the docs?  As long as they set some valid, stable directory it will all work...